contracts/Super1155.sol

```diff
@@ -648,10 +652,10 @@ contract Super1155 is PermitControl, ERC165Storage, IERC1155, IERC1155MetadataUR
     @param _groupId The ID of the item group to create or configure.
     @param _data The `ItemGroup` data input.
   */
-  function configureGroup(uint256 _groupId, ItemGroupInput calldata _data)
-    external virtual hasItemRight(_groupId, CONFIGURE_GROUP) {
+  function configureGroup(uint256 _groupId, ItemGroupInput calldata _data) external virtual {
     require(_groupId != 0,
       "Super1155: group ID 0 is invalid");
+    require(_hasItemRight(_groupId, CONFIGURE_GROUP), "Super1155: you don't have rights to configure group");
```

Comment:
Why did we make this a function call instead of a modifier? Is this for gas-efficiency reasons?

Reply:
Yes, having modifier with long condition chain add and several resolve paths adds huge bytecode overhead. Rewriting to require statement saved 2.5KB.

Reply:
> 
> 
> Yes, having modifier with long condition chain add and several resolve paths adds huge bytecode overhead. Rewriting to require statement saved 2.5KB.

Holy crap, never thought it could be that bad. RIP modifiers.